documents/coredocument.go

```diff
@@ -457,9 +504,82 @@ func (cd *CoreDocument) GetCollaborators(filterIDs ...identity.DID) ([]identity.
 	return filterCollaborators(cs, filterIDs...), nil
 }
 
+// GetCollaborators1 returns the collaborators excluding the filteredIDs
+func (cd *CoreDocument) GetCollaborators1(filterIDs ...identity.DID) (CollaboratorsAccess, error) {
+
+	rcs, err := cd.getReadCollaborators(coredocumentpb.Action_ACTION_READ_SIGN, coredocumentpb.Action_ACTION_READ)
+	if err != nil {
+		return CollaboratorsAccess{}, err
+	}
+	wcs, err := cd.getWriteCollaborators(coredocumentpb.TransitionAction_TRANSITION_ACTION_EDIT)
+	if err != nil {
+		return CollaboratorsAccess{}, err
+	}
+	rc := filterCollaborators(rcs, filterIDs...)
```

Comment:
these filter calls can be easily avoided by passing a filtered ids to the `getReadCollaborators` and  `getWriteCollaborators`.

Reply:
those methods don't take filtered ids as a parameter as of now... are you thinking of GetCollaborators?

Reply:
I mean you can modify them to take it as params, I don't know the complexity of changing them, but looks pretty straight forward here?